.github/workflows/book.yml

```diff
@@ -0,0 +1,38 @@
+# Performs checks, builds and deploys the mdbook
+
+name: book
+
+on:
+  push:
+    branches: [main, next]
```

Comment:
Am I missing it or do we not also need a similar workflow that runs on every PR to check if the docs can be built successfully? E.g. if I change the docs and insert an invalid link, I won't be notified until it is merged to `next` or `main` and the linkchecker complains.